homework-2/src/js/listeners-inputs.js

```diff
@@ -0,0 +1,51 @@
+const checkValidInput = (item) => {
+  item.getElementsByTagName('input')[0].addEventListener('change', (e) => {
+    const value = e.currentTarget.value;
+    const regul = new RegExp(
+      e.currentTarget.getAttribute('type') == 'email'
+        ? /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+        : /(?=.*[A-Z]+)(?=.*[0-9])(?=.*[a-z]){6,}/,
+    );
+
+    if (!regul.test(value)) {
```

Comment:
What about case when we have no value in our input?

Reply:
I apparently pushed not the last version
File "listeners-inputs" removed